app/src/main/java/com/xinto/mauth/ui/component/pinboard/PinBoard.kt

```diff
@@ -29,54 +31,140 @@ fun PinBoard(
     modifier: Modifier = Modifier,
     state: PinBoardState = rememberPinBoardState()
 ) {
-    FlowRow(
-        modifier = modifier,
-        verticalArrangement = Arrangement.spacedBy(8.dp, Alignment.CenterVertically),
-        horizontalArrangement = Arrangement.spacedBy(16.dp, Alignment.CenterHorizontally),
-        maxItemsInEachRow = 3
-    ) {
-        state.buttons.forEach { button ->
-            when (button) {
-                is PinBoardState.PinBoardButton.Number -> {
-                    PinButton(
-                        modifier = Modifier.weight(1f),
-                        onClick = { state.onNumberClick(button.number) }
-                    ) {
-                        Text(button.toString())
+    val orientation = LocalConfiguration.current.orientation
+    if (orientation == Configuration.ORIENTATION_LANDSCAPE) {
+
+        FlowColumn(
+            modifier = modifier,
+            verticalArrangement = Arrangement.spacedBy(8.dp, Alignment.CenterVertically),
+            horizontalArrangement = Arrangement.spacedBy(16.dp, Alignment.CenterHorizontally),
+            maxItemsInEachColumn = 4
+        ) {
+            val rowCount = 4
+            val rowWidth = 3
+            val columnCount = 3
+            val columnButtons = buildList {
```

Comment:
Currently I am transforming the rows into columns so we can use the FlowColumn + weight to make our buttons adjust the height using `weight(1f)`. There might be another way of doing it, but I am not aware.

Reply:
Don't forget to use `remember {}` when dealing with data. Otherwise that `buildList` will get executed on every rerender. I think `remember(columnCount, rowCount) {} should be fine, but those are constants anyway, so it's better to inline